apps/transport/lib/transport_web/live/on_demand_validation_select_live.ex

```diff
@@ -19,17 +19,20 @@ defmodule TransportWeb.Live.OnDemandValidationSelectLive do
      socket
      |> socket_data(%{
        select_options: select_options(),
-       changeset:
-         Ecto.Changeset.cast(
-           {%{url: "", type: "", feed_url: ""}, %{url: :string, type: :string, feed_url: :string}},
-           %{},
-           [:url, :type, :feed_url]
-         )
+       changeset: cast(%{})
      })}
   end
 
+  defp cast(params) do
```

Comment:
Suggestion : utiliser la fonction de Ecto.cast partout qui se charge de convertir les string en atoms, et qui trie les clés autorisées.

https://hexdocs.pm/ecto/Ecto.Changeset.html#cast/4

"Applies the given params as changes for the given data according to the given set of permitted keys. Returns a changeset.

The given data may be either a changeset, a schema struct or a {data, types} tuple. The second argument is a map of params that are cast according to the type information from data. params is a map with string keys or a map with atom keys, containing potentially invalid data. Mixed keys are not allowed.

During casting, all permitted parameters whose values match the specified type information will have their key name converted to an atom and stored together with the value as a change in the :changes field of the changeset. All parameters that are not explicitly permitted are ignored.
"